crates/bevy_asset/src/asset_changed.rs

```diff
@@ -0,0 +1,475 @@
+//! Filter query to limit iteration over [`Handle`]s which content was updated recently.
+use std::marker::PhantomData;
+
+use bevy_ecs::{
+    archetype::{Archetype, ArchetypeComponentId},
+    component::{ComponentId, Tick},
+    prelude::{Changed, Entity, Or, Resource, World},
+    query::{Access, FilteredAccess, QueryItem, ReadFetch, ReadOnlyWorldQuery, WorldQuery},
+    storage::{Table, TableRow},
+    world::unsafe_world_cell::UnsafeWorldCell,
+};
+use bevy_utils::{get_short_name, HashMap};
+
+use crate::{Asset, AssetId, Handle};
+
+#[doc(hidden)]
+#[derive(Resource)]
+pub struct AssetChanges<A: Asset> {
+    change_ticks: HashMap<AssetId<A>, Tick>,
+    last_change_tick: Tick,
+}
+
+impl<A: Asset> AssetChanges<A> {
+    pub(crate) fn insert(&mut self, asset_id: AssetId<A>, tick: Tick) {
+        self.last_change_tick = tick;
+        self.change_ticks.insert(asset_id, tick);
+    }
+    pub(crate) fn remove(&mut self, asset_id: &AssetId<A>) {
+        self.change_ticks.remove(asset_id);
+    }
+}
+impl<A: Asset> Default for AssetChanges<A> {
+    fn default() -> Self {
+        Self {
+            change_ticks: Default::default(),
+            last_change_tick: Tick::new(0),
+        }
+    }
+}
+
+struct AssetChangeCheck<'w, A: Asset> {
+    handle_change_map: &'w HashMap<AssetId<A>, Tick>,
```

Comment:
This is called `change_ticks` in the main data structure (`AssetChanges<A>`), why a different name `change_map` here? Also not sure the `handle_` prefix is really necessary.